cmake/dts.cmake

```diff
@@ -246,6 +246,18 @@ if(SUPPORTS_DTS)
   file(WRITE ${PROJECT_BINARY_DIR}/${BOARD}.dts_compiled
        "See zephyr.dts for the final merged devicetree.")
 
+  add_custom_target(
+    nrf_pin_report
+    DEPENDS
+    ${ZEPHYR_BASE}/scripts/dts/nrf_pin_report.py
+    COMMAND
+    ${PYTHON_EXECUTABLE}
+    ${ZEPHYR_BASE}/scripts/dts/nrf_pin_report.py
+    --dts ${ZEPHYR_DTS}
+    --bindings-dirs ${DTS_ROOT_BINDINGS}
+    USES_TERMINAL
+    )
```

Comment:
I think this is too nRF specific at current stage, to be a build target we want on all boards.
For example, having a `ninja nrf_pin_report` if my board is `sam4s_explained` is a bit weird.

Could we place this code in a dedicated cmake file, and only include it for nRF boards ?

Similar to how `boards/common/nrfjprog.board.cmake` is included ?

Reply:
> I think this is too nRF specific at current stage, to be a build target we want on all boards.

To be clear, this is just an RFC to show what we can do for Nordic.

That is why I wrote this in the description:


And for emphasis, I do not want to solve this for other boards. Nordic SoCs are the only ones whose devicetrees are used in this way to encode pin mux information and there are a lot of changes going on in other SoCs in this regard right now, so it's not stable.

This is why I named the target **nrf_pin_report** instead of something else.

Reply:
> To be clear, this is just an RFC to show what we can do for Nordic.

Ahh, right. I just looked at it as a PR, cause it was not a draft. My mistake.

> Now, the question is, is this worth having in mainline, just for Nordic SoCs and properly scoped and organized, at all, or not?

I think so, cause there could be people using Vanilla Zephyr with nordic boards.

Zephyr also contains dedicated CMake stuff for other vendors, such as:
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/xtensa/intel_apl_adsp/common/bootloader.cmake
https://github.com/zephyrproject-rtos/zephyr/blob/master/soc/xtensa/intel_apl_adsp/common/bootloader/CMakeLists.txt

but that is located in the soc folder.

So I was more thinking if we could located this in a Nordic dts folder or under https://github.com/zephyrproject-rtos/zephyr/tree/master/soc/arm/nordic_nrf.
Although it uses dts info for the report, then the user in principle should be concerned on how the information is gathered.